src/components/Filters/index.tsx

```diff
@@ -108,29 +110,49 @@ interface FiltersState {
 class Filters extends React.Component<FiltersProps, FiltersState> {
 	constructor(props: FiltersProps) {
 		super(props);
-		const { filters, schema, views } = this.props;
+		const { filters = [], schema, views = [] } = this.props;
 
 		const flatSchema = SchemaSieve.flattenSchema(schema);
+		const flatViews = this.flattenViews(views);
+		const flatFilters = filters.map(filter =>
+			SchemaSieve.flattenSchema(filter),
+		);
 
 		this.state = {
 			showModal: false,
 			searchString: '',
 			editingFilter: null,
-			filters: filters || [],
-			views: views || [],
+			filters: flatFilters,
+			views: flatViews,
 			schema: flatSchema,
 			edit: [],
 		};
 
 		this.state.edit.push(this.getCleanEditModel());
 	}
 
-	componentWillReceiveProps(prevProps: FiltersProps) {
+	componentWillReceiveProps(nextProps: FiltersProps) {
+		const newState: any = {};
```

Comment:
Could this be types as `Partial<FiltersState>`?

Reply:
I tried this and it turned into a bit of a nightmare trying to make the compiler happy!

Reply:
Maybe `{ [key: string]: any }` could be slightly better.